src/Fable.Transforms/AST/AST.Babel.fs

```diff
@@ -212,481 +255,477 @@ type MacroExpression(value, args, ?loc) =
+        member _.Print(printer) =
+            // TODO: optional, type annotation
+            printer.Print(name, ?loc=loc)
 
 // Literals
 type RegExpLiteral(pattern, flags_, ?loc) =
-    inherit Literal("RegExpLiteral", ?loc = loc)
     let flags =
         flags_ |> Seq.map (function
             | RegexGlobal -> "g"
             | RegexIgnoreCase -> "i"
             | RegexMultiline -> "m"
             | RegexSticky -> "y") |> Seq.fold (+) ""
-    member __.Pattern: string = pattern
-    member __.Flags: string = flags
-    override _.Print(printer) =
-        printer.Print("/", ?loc=loc)
-        printer.Print(pattern)
-        printer.Print("/")
-        printer.Print(flags)
+    member _.Pattern: string = pattern
+    member _.Flags: string = flags
+    interface Literal with
+        member _.Print(printer) =
+            printer.Print("/", ?loc=loc)
+            printer.Print(pattern)
+            printer.Print("/")
+            printer.Print(flags)
 
 type Undefined(?loc) =
-    inherit Identifier("undefined", ?loc = loc)
     // TODO: Use `void 0` instead? Just remove this node?
-    override _.Print(printer) =
-        printer.Print("undefined", ?loc=loc)
+    interface Expression with
+        member _.Print(printer) =
+            printer.Print("undefined", ?loc=loc)
 
 type NullLiteral(?loc) =
-    inherit Literal("NullLiteral", ?loc = loc)
-    override _.Print(printer) =
-        printer.Print("null", ?loc=loc)
+    interface Literal with
+        member _.Print(printer) =
+            printer.Print("null", ?loc=loc)
 
 type StringLiteral(value, ?loc) =
-    inherit Literal("StringLiteral", ?loc = loc)
-    member __.Value: string = value
-    override _.Print(printer) =
-        printer.Print("\"", ?loc=loc)
-        printer.Print(System.Web.HttpUtility.JavaScriptStringEncode(value))
-        printer.Print("\"")
+    member _.Value: string = value
+    interface Literal with
+        member _.Print(printer) =
+            printer.Print("\"", ?loc=loc)
+            printer.Print(printer.EscapeJsStringLiteral(value))
+            printer.Print("\"")
 
 type BooleanLiteral(value, ?loc) =
-    inherit Literal("BooleanLiteral", ?loc = loc)
-    member __.Value: bool = value
-    override _.Print(printer) =
-        printer.Print((if value then "true" else "false"), ?loc=loc)
+    member _.Value: bool = value
+    interface Literal with
+        member _.Print(printer) =
+            printer.Print((if value then "true" else "false"), ?loc=loc)
 
 type NumericLiteral(value, ?loc) =
-    inherit Literal("NumericLiteral", ?loc = loc)
-    member __.Value: float = value
-    override _.Print(printer) =
-        printer.Print(value.ToString(), ?loc=loc)
+    member _.Value: float = value
+    interface Literal with
+        member _.Print(printer) =
+            let value =
+                match value.ToString() with
+                | "∞" -> "Infinity"
+                | "-∞" -> "-Infinity"
+                | value -> value
```

Comment:
This is the best way I could find to print numbers but it doesn't look very robust. Please feel free to edit if you can think of a better way @ncave